clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -5150,6 +5150,10 @@ void Sema::InstantiateVariableDefinition(SourceLocation PointOfInstantiation,
   if (TSK == TSK_ExplicitSpecialization)
     return;
 
+  if (getLangOpts().SYCLIsDevice && !Var->hasAttr<SYCLKernelAttr>() &&
+      !Var->hasAttr<SYCLDeviceAttr>())
+    return;
```

Comment:
This stuff doesn't seem right. That is probably why tests are failing.
`sycl_kernel` attribute can be applied only to a function template.
Only `SYCL_EXTERNAL` functions can have `sycl_device` attribute.